pkg/kp/rcstore/store.go

```diff
@@ -0,0 +1,32 @@
+package rcstore
+
+import (
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/rc/fields"
+)
+
+// Store represents an interface for persisting replication controllers to Consul,
+// as well as restoring replication controllers from Consul.
+type Store interface {
+	// Create creates a replication controller with the specified manifest and selectors.
+	// The node selector is used to determine what nodes the replication controller may schedule on.
+	// The pod label set is:
+	// 1) applied to every pod the replication controller schedules,
+	// 2) used (in addition to replication controller ID) to determine what pods are already scheduled.
+	Create(manifest pods.Manifest, nodeSelector labels.Selector, podLabels labels.Set) (fields.RC, error)
+
+	List() ([]fields.RC, error)
+	SetDesiredReplicas(fields.RC, int) error
```

Comment:
using an ID now, typedefed it in case anything changes, but I don't think anything will